neal/sampler.py

```diff
@@ -239,21 +254,33 @@ def sample(self, bqm, beta_range=None, num_reads=None, num_sweeps=1000,
         if interrupt_function and not callable(interrupt_function):
             raise TypeError("'interrupt_function' should be a callable")
 
+        assert num_sweeps_per_beta >= 1, "'num_sweeps_per_beta' must be a positive whole value"
+        
         # handle beta_schedule et al
-        if beta_range is None:
-            beta_range = _default_ising_beta_range(bqm.linear, bqm.quadratic)
-
-        num_sweeps_per_beta = max(1, num_sweeps // 1000.0)
-        num_betas = int(math.ceil(num_sweeps / num_sweeps_per_beta))
-        if beta_schedule_type == "linear":
-            # interpolate a linear beta schedule
-            beta_schedule = np.linspace(*beta_range, num=num_betas)
-        elif beta_schedule_type == "geometric":
-            # interpolate a geometric beta schedule
-            beta_schedule = np.geomspace(*beta_range, num=num_betas)
+        if beta_schedule_type == "custom":
+            assert len(beta_schedule)>0, "'beta_schedule' must be provided for beta_schedule_type = 'custom'"
+            beta_schedule = np.array(beta_schedule,dtype=float)
         else:
-            raise ValueError("Beta schedule type {} not implemented".format(beta_schedule_type))
-
+            assert len(beta_schedule) == 0, "'beta_schedule' must be set to default (=[]) unless 'beta_schedule_type' is custom"
+            assert num_betas >= 1, "'num_betas' must be a positive value"
+            if beta_range == None:
+                beta_range = _default_ising_beta_range(bqm.linear, bqm.quadratic)
+            else:
+                assert len(beta_range) == 2 and beta_range[0] >= 0 and beta_range[1] >= beta_range[0], "'beta_range' should be increasing list 2 positive values"
```

Comment:
Do you mean here "of at least two positive values"?

Reply:
beta_range (as opposed to beta_schedule) must contain exactly 2 values if provided.